src/journal/ObjectRecorder.cc

```diff
@@ -412,17 +412,18 @@ void ObjectRecorder::send_appends_aio() {
     client::guard_append(&op, m_soft_max_size);
     auto append_buffers = &m_in_flight_appends[append_tid];
 
+    bufferlist bl;
     for (auto it = m_pending_buffers.begin(); it != m_pending_buffers.end(); ) {
       ldout(m_cct, 20) << __func__ << ": flushing " << *it->first << dendl;
-      op.append(it->second);
-      op.set_op_flags2(CEPH_OSD_OP_FLAG_FADVISE_DONTNEED);
+      bl.append(it->second);
       m_aio_sent_size += it->second.length();
       append_buffers->push_back(*it);
       it = m_pending_buffers.erase(it);
       if (m_aio_sent_size >= m_soft_max_size) {
         break;
       }
     }
+    client::append(&op, bl);
```

Comment:
We can only do this if the min OSD version is >= octopus, otherwise you need to fallback to the regular RADOS append op.